mlir-sycl/lib/Conversion/SYCLToLLVM/SYCLToLLVM.cpp

```diff
@@ -641,4 +700,6 @@ void mlir::sycl::populateSYCLToLLVMConversionPatterns(
   if (typeConverter.getOptions().useBarePtrCallConv)
     patterns.add<BarePtrCastPattern>(typeConverter, /*benefit*/ 2);
   patterns.add<ConstructorPattern>(typeConverter);
+  if (typeConverter.getOptions().useBarePtrCallConv)
```

Comment:
What happens when useBarePtrCallConv is not given? Are we unable to lower to LLVM?

Reply:
Yes. Our lowering has always depended on this option being passed. The only difference is that we make it explicit here. I wanted to discuss it next meeting (I think we should assert it is set), but we can do it like this for now.

Reply:
Ok, let's discuss it on our next meeting. I also think we should assert it is set, as our lowering depends on it.